jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/events/JettyAnnotatedEventDriver.java

```diff
@@ -115,11 +114,11 @@ else if (events.onBinary.isStreaming())
                     }
                     catch (Throwable t)
                     {
-                        // dispatched calls need to be reported
                         session.close(t);
+                        return;
```

Comment:
Wouldn't this return leave a `MessageInputStream` laying around in a different thread?

Reply:
Not sure what you mean here, this is in the same thread the `MessageInputStream` is being read from. So `onMessage` will have thrown so now we need to close the `Session`. 

If we don't return and call `handlerComplete()`, this will try to resume reading more frames but we have already closed the session. We could close the `MessageInputStream` before returning but this will not have a major effect as no one is reading from it.